src/db/collections/outputs/indexer/mod.rs

```diff
@@ -75,16 +76,11 @@ impl OutputCollection {
         ledger_index: MilestoneIndex,
     ) -> Result<Option<IndexedOutputResult>, Error> {
         let id = id.into();
-        let id_string = match id {
-            IndexedId::Alias(_) => "output.alias_id",
-            IndexedId::Foundry(_) => "output.foundry_id",
-            IndexedId::Nft(_) => "output.nft_id",
-        };
         let mut res = self
             .aggregate(
                 [
                     doc! { "$match": {
-                        id_string: id,
+                        "output.details.indexed_id": id,
```

Comment:
I'm not sure this is correct. The indexer API differentiates between NFTs, aliases, and foundries. As far as I can tell we never check if the output returned from the database matches the queried output type.

Reply:
Those ids are unique across NFTs, Aliases and Foundries ... so querying for a specific id will return the right output type (as long as we stored/computed the explicit id correctly), right?

Reply:
That, and also we check the `output.kind` in our queries anyway.

Reply:
I'm talking about the `indexed_output_by_id` query in particular. If a user requests an alias output by using `/api/indexer/v1/outputs/alias/{aliasId}` but for some reason provides an `foundryId`, the response should be either `400` or `404`.

Reply:
you're right... I suggest adding:
```rust
impl IndexedId {
    fn kind(&self) -> &str {
        match self {
            Self::Alias(_) => AliasOutput::KIND,
            Self::Foundry(_) => FoundryOutput::KIND,
            Self::Nft(_) => NftOutput::KIND,
        }
    }
}
```
to further restrict the `$match`.